js/main.js

```diff
@@ -28,6 +35,117 @@ function addButtonHandler(event) {
   data.nextEntryId++;
   data.entries.unshift(journalEntry);
 
+  // Resetting the form after parsing
   $journalForm.reset();
   $image.setAttribute('src', 'images/placeholder-image-square.jpg');
+
+  // Rendering the entry and prepending it to the list
+  $divDataViewEntries.prepend(renderEntry(journalEntry));
+
+  viewSwap($entriesButton); // Using my parsed entries button here makes sure the argument i give the function matches what it is expecting.
+
+  toggleNoEntries();
+}
+
+function renderEntry(entry) {
+  const $divRow = document.createElement('div');
+  const $divColumn = document.createElement('div');
+  const $divColumn2 = document.createElement('div');
+  const $liEntry = document.createElement('li');
+  const $imgEntryImage = document.createElement('img');
+  const $divTitleList = document.createElement('div');
+  const $h2EntryTitle = document.createElement('h2');
+  const $divNotesEntry = document.createElement('div');
+  const $pNotes = document.createElement('p');
+
+  $divRow.className = 'row';
+  $divColumn.className = 'column-half';
+  $divColumn2.className = 'column-half';
+  $imgEntryImage.className = 'image';
+  $imgEntryImage.setAttribute('src', entry.photoUrl);
+  $h2EntryTitle.textContent = entry.title;
+  $pNotes.textContent = entry.notes;
+
+  $divRow.appendChild($divColumn);
+  $divRow.appendChild($divColumn2);
+  $divColumn.appendChild($imgEntryImage);
+  $divColumn2.appendChild($liEntry);
+  $liEntry.appendChild($divTitleList);
+  $liEntry.appendChild($divNotesEntry);
+  $divTitleList.appendChild($h2EntryTitle);
+  $divNotesEntry.appendChild($pNotes);
+
+  /*  This function creates the following dom tree in which a journal entry's properties are displayed:
+              <li>
+              <div class ="row">
+              <div class = "column half">
+                <img
+                  class="image"
+                  src=""
+                  alt="" />
+                </div>
+                <div class = "column-half">
+                <div>
+                  <h2>title goes here</h2>
+                </div>
+                <div>
+                  <p>Notes go here</p>
+                </div>
+                </div>
+              </li>
+        */
+  return $divRow;
```

Comment:
Your outtermost element should be an `li`. So you should be `return`ing your `li`